Update snowflake-configs.md (#7081)
fishtown-analytics/docs.getdbt.com: website/docs/reference/resource-configs/snowflake-configs.md

- Append
- Delete+insert
- Insert_overwrite

>   - Note: This is not a standard dbt incremental strategy. `insert_overwrite` behaves like `truncate` + re-`insert` commands on Snowflake. It doesn't support partition-based overwrites, which means it'll overwrite the entire table intentionally. We have implemented it as an incremental strategy because it aligns with the dbt workflow of not dropping the existing table. 

Reviewer: ```suggestion
  - Note: This is not a standard dbt incremental strategy. `insert_overwrite` behaves like `truncate` + re-`insert` commands on Snowflake. It doesn't support partition-based overwrites, which means it'll overwrite the entire table intentionally. It's implemented as an incremental strategy because it aligns with dbt’s workflow of not dropping existing tables.
```
